controllers/nodehealthcheck_controller.go

```diff
@@ -417,18 +430,19 @@ func (r *NodeHealthCheckReconciler) isHealthy(conditionTests []remediationv1alph
 			now := currentTime()
 			if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
 				// unhealthy condition duration expired, node is unhealthy
-				return false, nil
+				unhealthyNodeReason := fmt.Sprintf("type:%s, status:%s, duration(max):%v", c.Type, c.Status, c.Duration.Duration)
```

Comment:
- to keep the label value cardinality lower, I would skip the duration. It's not important IMHO, because it's known, it's in the NHC CR
- for the same reason we probably can also skip the status?
- when only type is left, maybe `unhealthyConditionType` is a better name than "reason" everywhere?
- what if multiple conditions match? At once. Or one after another but overlappingin time. Wasn't relevant so far...

Reply:
> I would skip the duration

I agree, in case of multiple NHC with different values (if this is possible), we can still differentiate by instance

> for the same reason we probably can also skip the status?

shouldn't it be useful to differentiate between "Ready" and "Unknown" statuses?

> what if multiple conditions match? At once. Or one after another but overlapping in time. Wasn't relevant so far...

Not sure if I got it, but it stores the "reason" into the CR, is it possible to have multiple CRs for the same node for different unmet conditions?

Reply:
> shouldn't it be useful to differentiate between "Ready" and "Unknown" statuses?

not sure, I tend to say it's not worth adding complexity to the metric 🤔 

>  is it possible to have multiple CRs for the same node for different unmet conditions

no, that's the point: so far we didn't care WHY a node is unhealthy (which condition caused it), just that it IS unhealthy. Now the metric cares about the why, which is a difference. The why can be multiple unhealthy conditions.

Example: NHC looks for storage and network related conditions. First storage fails, Node gets unhealthy, CR with storage reason annotation is created. Then the node also gets a network problem. CR keeps storage reason annotation only? Then storage is ok, network still fails. What about the CR? Then network is ok, CR deleted. What reason is tracked for the metric?

Reply:
> no, that's the point: so far we didn't care WHY a node is unhealthy (which condition caused it), just that it IS unhealthy. Now the metric cares about the why, which is a difference. The why can be multiple unhealthy conditions.

I see now, basically what the metric tracks is actually the reason behind the creation of the CR, more than the "unhealthiness". So why NHC intervened, more than the actual node problem.